src/Extension/SmartPunct/PunctuationParser.php

```diff
@@ -18,32 +18,29 @@
 
 use League\CommonMark\Node\Inline\Text;
 use League\CommonMark\Parser\Inline\InlineParserInterface;
+use League\CommonMark\Parser\Inline\InlineParserMatch;
 use League\CommonMark\Parser\InlineParserContext;
 
 final class PunctuationParser implements InlineParserInterface
 {
-    /**
-     * {@inheritdoc}
-     */
-    public function getCharacters(): array
+    public function getMatchDefinition(): InlineParserMatch
     {
-        return ['-', '.'];
+        return InlineParserMatch::oneOf('-', '.');
     }
 
-    public function parse(InlineParserContext $inlineContext): bool
+    public function parse(string $match, InlineParserContext $inlineContext): bool
     {
         $cursor = $inlineContext->getCursor();
-        $ch     = $cursor->getCharacter();
 
         // Ellipses
-        if ($ch === '.' && $matched = $cursor->match('/^\\.( ?\\.)\\1/')) {
+        if ($match === '.' && $matched = $cursor->match('/^\\.( ?\\.)\\g{-1}/')) {
```

Comment:
`g{-1}` is a relic of an earlier attempt using named subgroups to match parsers and is no longer needed here